File: android/mekuri/src/main/kotlin/app/mekuri/MekuriZone.kt
package app.mekuri

/**
 * Horizontal region of the container a touch lands in. Each edge zone spans
 * `tapZoneRatio` of the width; points on a boundary fall in [Center].
 */
internal enum class MekuriZone {
    Leading,
    Center,
    Trailing,

    ;

    internal companion object {
        internal fun resolve(x: Float, width: Float, configuration: MekuriConfiguration): MekuriZone {
            val edge = width * configuration.tapZoneRatio
            return when {
                x < edge -> Leading
                x > width - edge -> Trailing
                else -> Center
            }
        }
    }
}

File: ios/Sources/Mekuri/MekuriZone.swift
import CoreGraphics

/// Horizontal region of the page a touch lands in. Each edge zone spans
/// `tapZoneRatio` of the width; points on a boundary fall in the centre.
enum MekuriZone: Equatable, Sendable {
    case leading
    case center
    case trailing

    static func resolve(x: CGFloat, width: CGFloat, configuration: MekuriConfiguration) -> MekuriZone {
        let edge = width * configuration.tapZoneRatio
        if x < edge { return .leading }
        if x > width - edge { return .trailing }
        return .center
    }
}
